protocol/contracts/fakes/templegold/TempleGoldMock.sol

```diff
@@ -0,0 +1,173 @@
+pragma solidity ^0.8.20;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Temple (templegold/TempleGoldMock.sol)
+
+
```

Comment:
Why is this needed? Why cant you use the real one?

Reply:
for `TempleGoldMock`, i only test cross-chain transfers with LZ. and the complexities of minting and distribution, i wanted to obfuscate. that's why I used a trimmed version of `TempleGold` as mock for only cross-chain transfer. Still think I should use the full version?

Will revert for `TempleTokenMock`

Reply:
My view generally is that we should be testing against the contracts we're going to release